vlm/src/main/scala/GeoTiffRasterSource.scala

```diff
@@ -44,47 +44,76 @@ class GeoTiffRasterSource(val fileURI: String) extends RasterSource {
   def bandCount: Int = tiff.bandCount
   def cellType: CellType = tiff.cellType
 
+  def withCRS(targetCRS: CRS, resampleMethod: ResampleMethod = NearestNeighbor): GeoTiffRasterSource =
+    new GeoTiffRasterSource(fileURI) {
```

Comment:
Making anonymous classes like this should be avoided from code hygiene perspective. In the stack trace they're not going to have any meaningful name and it makes it not totally obvious where important code lives. This looks like `WarpGeoTiffRasterSource` from https://github.com/geotrellis/geotrellis-contrib/pull/11/files ? 

Either that PR should be merged and rebase or this class should get a name.

Reply:
I think it can be merged as is, as after adding the unit test that compares this source to the common geotrellis tiling code it at least works as expected, after that we can combine it with your @echeipesh PR, what do you think?

Reply:
That sounds good to me